crates/iota-protocol-config/src/lib.rs

```diff
@@ -1448,7 +1448,12 @@ impl ProtocolConfig {
     }
 
     pub fn score_based_rewards(&self) -> bool {
-        self.feature_flags.score_based_rewards
+        let score_based_rewards = self.feature_flags.score_based_rewards;
+        assert!(
+            score_based_rewards || self.scorer_version.is_none(),
```

Comment:
```suggestion
            !score_based_rewards || self.scorer_version.is_some(),
```
Just to double-check the condition, from the related message, it looks like we need to check that `scorer_version` is set when `score_based_rewards` is enabled.

Reply:
You're right, thanks for catching that.